lib/utils.ts

```diff
@@ -59,10 +59,35 @@ export function generateBuildTasks(
 				img.image.dockerfile != null
 					? { dockerfilePath: img.image.dockerfile }
 					: {},
+				// Pass through args, context, labels, tag.
+				//
 				// It's possible to specify an image name as well as
 				// a build, but that doesn't make sense in a balena
 				// ecosystem, so we remove it
-				_.omit(img.image, 'image'),
+				//
+				// We also remove anything that goes into dockerOpts in the next step
+				_.omit(
+					img.image,
+					'image',
+					'cache_from',
+					'extra_hosts',
+					'network',
+					'shm_size',
+					'target',
+				),
```

Comment:
No need to use lodash, you could do:

```typescript
const {
  image,
  cache_from,
  extra_hosts,
  network,
  shm_size,
  target,
  ...restImageProps
} = img.image
```

`restImageProps` is an object containing all props except keys above. You could use `target`, `shm_size`... in code below.